qualang_tools/optimal_weights/StateDiscriminator.py

```diff
@@ -0,0 +1,256 @@
+import numpy as np
+from pandas import DataFrame
+from qm.qua import *
+import os
+import matplotlib.pyplot as plt
+from sklearn import mixture
+from scipy import signal
+
```

Comment:
We need to see that all of these packages are in the environment. Did you check that the code works in a fresh env?

Reply:
After installing qm-qua and qualang-tools in a fresh env: `from sklearn import mixture` and import seaborn as sns` are missing

Reply:
Yes, these needs to be added to qualang tools...

Reply:
It was a package in the simulation programs but not part of the tools, so not sure why it was originally there